mm2src/mm2_main/src/lp_price.rs

```diff
@@ -205,36 +207,29 @@ pub struct CEXRates {
     pub rel: BigDecimal,
 }
 
-// Fetcher funtion to test fetching latest prices from a endoint.
+/// fetcher funtion to try fetch latest price from single endoint.
 async fn try_price_fetcher_endpoint(
     endpoint: &str,
     base: &str,
     rel: &str,
 ) -> Result<CEXRates, MmError<PriceServiceRequestError>> {
-    match process_price_request(endpoint).await {
-        Ok(response) => match response.get_cex_rates(base.to_string(), rel.to_string()) {
-            Some(fiat_price) => {
-                let (base_usd_price, rel_usd_price) = fiat_price.get_rate_price();
-                Ok(CEXRates {
-                    base: base_usd_price,
-                    rel: rel_usd_price,
-                })
-            },
-            None => {
-                let cex: Option<CEXRates> = None;
-                cex.or_mm_err(|| PriceServiceRequestError::Internal("Couldn't fetch price".to_string()))
-            },
-        },
-        Err(e) => MmError::err(e.into_inner()),
-    }
+    let response = process_price_request(endpoint).await?;
+    let fiat_price = response
+        .get_cex_rates(base.to_string(), rel.to_string())
```

Comment:
Maybe change `get_cex_rates` fn to accept `base: &str` and `rel: &str` instead of `String`, this will reduce some clones.